node.go

```diff
@@ -18,11 +18,28 @@ import (
 
 const CBORTagLink = 42
 
-func init() {
-	// Register the block decoder
-	node.DefaultBlockDecoder[cid.DagCBOR] = func(b blocks.Block) (node.Node, error) { return DecodeBlock(b) }
+// Decode a CBOR encoded Block into an IPLD Node.
+//
+// This method *does not* canonicalize and *will* preserve the CID. As a matter
+// of fact, it will assume that `block.Cid()` returns the correct CID and will
+// make no effort to validate this assumption.
+//
+// In general, you should not be calling this method directly. Instead, you
+// should be calling the `Decode` method from the `go-ipld-format` package. That
+// method will pick the right decoder based on the Block's CID.
+//
+// Note: This function keeps a reference to `block` and assumes that it is
+// immutable.
+func DecodeBlock(block blocks.Block) (node.Node, error) {
```

Comment:
Note: I moved this function to the top and changed the return type to `node.Node` (instead of `cbornode.Node`) as this is the main public interface.